src/main/kotlin/no/liflig/documentstore/dao/SearchRepositoryWithCount.kt

```diff
@@ -0,0 +1,159 @@
+ */
+abstract class AbstractSearchRepositoryWithCount<EntityIdT, EntityT, SearchQueryT>(
+  jdbi: Jdbi,
+  sqlTableName: String,
+  serializationAdapter: SerializationAdapter<EntityT>,
+) :
+  SearchRepositoryWithCount<EntityIdT, EntityT, SearchQueryT>,
+  AbstractSearchRepository<EntityIdT, EntityT, SearchQueryT>(
+    jdbi, sqlTableName, serializationAdapter
+  ) where
+EntityIdT : EntityId,
+EntityT : EntityRoot<EntityIdT> {
+  private fun fromJson(value: String): EntityT = serializationAdapter.fromJson(value)
+
+  protected open val rowMapperWithCount =
+    createRowMapperWithCount(createRowParserWithCount(::fromJson))
+
+  /**
+   * Gets database objects matching the given parameters, and the total count of objects in the database.
+   *
+   * This can be used for pagination: for example, if passing e.g. `limit = 10` to display 10 items in a page at a time,
+   * the count can be used to display the number of pages.
+   */
+  protected open fun getByPredicateWithCount(
+    sqlWhere: String = "TRUE",
+    limit: Int? = null,
+    offset: Int? = null,
+    orderBy: String? = null,
+    orderDesc: Boolean = false,
+    bind: Query.() -> Query = { this }
+  ): EntitiesWithCount<EntityT> = mapExceptions {
+    val transaction = transactionHandle.get()
+
+    if (transaction != null) {
+      innerGetByPredicateWithCount(sqlWhere, transaction, limit, offset, orderBy, orderDesc, bind)
+    } else {
+      jdbi.open().use { handle ->
+        innerGetByPredicateWithCount(sqlWhere, handle, limit, offset, orderBy, orderDesc, bind)
+      }
+    }
+  }
+
+  private fun innerGetByPredicateWithCount(
+    sqlWhere: String,
+    handle: Handle,
+    limit: Int? = null,
+    offset: Int? = null,
+    orderBy: String? = null,
+    desc: Boolean = false,
+    bind: Query.() -> Query = { this }
+  ): EntitiesWithCount<EntityT> {
+    val limitString = limit?.let { "LIMIT $it" } ?: ""
+    val offsetString = offset?.let { "OFFSET $it" } ?: ""
+    val orderDirection = if (desc) "DESC" else "ASC"
+    val orderByString = orderBy ?: "created_at"
+
+    val rows =
+      handle
+        .select(
+          // SQL query based on https://stackoverflow.com/a/28888696
+          // Uses a RIGHT JOIN with the count in order to still get the count when no rows are
+          // returned.
+          """
+              WITH base_query AS (
+                  SELECT id, data, version, created_at, modified_at
+                  FROM "$sqlTableName"
+                  WHERE ($sqlWhere)
+              )
+              SELECT id, data, version, created_at, modified_at, count
+              FROM (
+                  TABLE base_query
+                  ORDER BY $orderByString $orderDirection
+                  $limitString
+                  $offsetString
+              ) sub_query
+              RIGHT JOIN (
+                  SELECT count(*) FROM base_query
+              ) c(count) ON true
```

Comment:
Man kan forenkle denne litt med bruk av ``over()``. Resten kopierte jeg fra den orginale ``GetByPredicate``, da jeg antok at den ellers var lik
```suggestion
 SELECT id, data, version, created_at, modified_at, count(1) OVER() AS count
              FROM "$sqlTableName"
              WHERE ($sqlWhere)
              ORDER BY $orderByString $orderDirection
              $limitString
              $offsetString
```

Reply:
Vurderte `over()` da jeg skrev queryet, men som beskrevet i kommentaren over her vil det ikke fungere som forventa når ingen rader returneres.